webpack.config.js

```diff
@@ -0,0 +1,169 @@
+import path from 'path';
+import { fileURLToPath } from 'url';
+import ReactRefreshPlugin from '@pmmmwh/react-refresh-webpack-plugin';
+import plugin from 'eslint-plugin-react';
+import HtmlWebpackPlugin from 'html-webpack-plugin';
+import MiniCssExtractPlugin from 'mini-css-extract-plugin';
+// Сначала определяем __filename и __dirname
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const srcPath = path.resolve(__dirname, 'src');
+import Dotenv from 'dotenv-webpack';
+import TsCheckerPlugin from 'fork-ts-checker-webpack-plugin';
+// Теперь можно использовать __dirname для определения buildPath
+const buildPath = path.resolve(__dirname, 'dist');
+
+const isProd = process.env.NODE_ENV === 'production';
+const getSettingsForStyles = (withModules = false) => {
+  // Заменяем в нашей функции style-loader на mini-css-extract-plugin
+  return [
+    isProd ? MiniCssExtractPlugin.loader : 'style-loader',
+    !withModules
+      ? 'css-loader'
+      : {
+          loader: 'css-loader',
+          options: {
+            modules: {
+              localIdentName: !isProd ? '[path][name]__[local]' : '[hash:base64]',
+              namedExport: false,
+            },
+          },
+        },
+    {
+      loader: 'postcss-loader',
+      options: {
+        postcssOptions: {
+          plugins: ['autoprefixer'],
+        },
+      },
+    },
+    {
+      loader: 'sass-loader',
+      options: {
+        additionalData: `
+              @use "@styles/styles.scss" as *;
+              @use "@styles/_fonts.scss" as *;
+              @use "@styles/_reset.scss" as *;
+              @use "@styles/_mixin.scss" as *;
+            `,
+      },
+    },
+  ];
+};
+export default {
+  entry: path.resolve(__dirname, 'src/main.tsx'),
+  target: !isProd ? 'web' : 'browserslist',
+  devtool: isProd ? 'hidden-source-map' : 'eval-source-map',
+  output: {
+    path: buildPath,
+    filename: 'bundle.js',
+  },
+
+  plugins: [
+    new HtmlWebpackPlugin({
+      template: path.resolve(__dirname, 'src/index.html'), // путь до нашего шаблона
+    }),
+    new Dotenv(),
+    !isProd && new ReactRefreshPlugin(),
+    //Добавим плагин в plugins
+    isProd &&
+      new MiniCssExtractPlugin({
+        // Для того чтобы файл со стилями не кэшировался в браузере добавим filename
+        filename: '[name]-[hash].css',
+      }),
+  ].filter(Boolean),
+  module: {
+    rules: [
+      {
+        test: /\.(ts|tsx)$/,
+        use: {
+          loader: 'babel-loader',
+    
+        },
+        exclude: /node_modules/, // Рекомендуется исключить node_modules
+      },
+      {
+        test: /\.svg$/,
+        use: '@svgr/webpack',
+        exclude: /node_modules/,
+      },
```

Comment:
[+] сейчас подключены только svg как реакт-компоненты, но в некоторых случаях свг используются как картинки (например, на странице рецепта) и поэтому они не загружаются

поэтому нужно добавить отдельный обработчик по примеру того, что используется ниже для png и jpg. а отличать эти два типа свг можно по приписке ?react при импорте, как использовалось для сборки на vite. в вебпаке это можно сделать с помощью `resourceQuery`:
```suggestion
     {
        test: /\.svg$/,
        use: '@svgr/webpack',
        resourceQuery: /react/, // include if *.svg?react
        exclude: /node_modules/,
      },
```